src/tak/Game.java

```diff
@@ -208,6 +218,18 @@ void resign(Player p) {
         whenGameEnd();
     }
     
+    void setMarked(boolean marked, String field)
+    {
+        int file = field.charAt(0) - 'a';
+        int rank = field.charAt(1) - 1;
+        this.board[rank][file].setMarked(marked);
```

Comment:
A7 will be out of bounds for 4x4 board
On Apr 24, 2016 4:55 PM, "TreffnonX" <notifications@github.com> wrote:

> In src/tak/Game.java
> <https://github.com/chaitu236/TakServer/pull/9#discussion_r60852700>:
>
> > @@ -208,6 +218,18 @@ void resign(Player p) {
> >          whenGameEnd();
> >      }
> >
> > +    void setMarked(boolean marked, String field)
> > +    {
> > +        int file = field.charAt(0) - 'a';
> > +        int rank = field.charAt(1) - 1;
> > +        this.board[rank][file].setMarked(marked);
>
> Happens in the call, but can be included here, of course.
> The regex only runs with [A-Ha-h] and [1-8].
> But i will add it for security's sake.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/chaitu236/TakServer/pull/9/files/c36a1d4254fa93b62999aa0c99ce83b75485f719#r60852700>
>